walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java

```diff
@@ -30,4 +30,24 @@ public void testGreeting() {
 
     Assert.assertEquals("Hello Ada", greeting);
   }
+
+  @Test
+  public void testGreetingTrimsWhitespace() {
+    Greeter greeter = new Greeter();
+
+    String greeting = greeter.greet("   Ada   ");
+
+    // Whitespace should be trimmed
+    Assert.assertEquals("Hello Ada", greeting);
+  }
+
+  @Test
+  public void testGreetingWithSpecialCharacters() {
+    Greeter greeter = new Greeter();
+
+    String greeting = greeter.greet("    @$Ada#%    ");
+
+    //Whitespace and Special characters should be removed
+    Assert.assertEquals("Hello Ada", greeting);
+  }
```

Comment:
Consider adding a test for someone with spaces in their first name.

What if my first name was "Anna Beth". This would result in "Hello AnnaBeth" right? is that how you intend it to work?

Reply:
I didn't really dwell much on it because it was shown as an example for the training but I improved it to incorporate the test case. Thanks for the suggestion!

Reply:
Awesome!